pkg/util/annotationkey.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+const (
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set
+	// ... Annotations are key/value pairs. Valid annotation keys have two segments: an optional prefix and name,
+	// separated by a slash (/). The name segment is required and must be 63 characters or less
+	K8sAnnotationNameLength = 63
+)
+
+// return vmName based key if length is not exceeing
+// otherwise combine UID and vmName for better troubleshooting
+//
+//	when the length of string after `/` is 63, then it is UID+vmName
+//	when the length of string after `/` is < 63, then it is only vmName
+func GenerateAnnotationKeyMigratingVM(vmName, vmUID string) string {
+	if len(AnnotationMigratingNamePrefix)+len(vmName) >= K8sAnnotationNameLength {
+		name := AnnotationMigratingNamePrefix + vmUID + "-" + vmName
```

Comment:
we could just simplify the logic and use `vmUID` only and avoid the `vmName` in the name. This would make scaling down later easier as we could just check if pattern is name or uid and identify accordingly

Reply:
The main concern here is the upgrade path.

Suppose when Harvester POD is upgrading, some VMs are migrated actively/passively in parallel, the vmname is booked, after Harvester POD is in new version:

if it assumes the booked info is UID, it will be wrong; 
if it check the booked info format is/not UID format, there is still a corner case that: a VM's name is specially defined with UID format, it may break the logic

If a VM's info it left on the resourcequota annotations, user needs to manually remove it.

The complex naming tries to avoid such.

Reply:
Another solution is to add a new annotation name like below, and also handle the previous vmname based annotation

`kubevirt.io/migrationJobUID: 74e7d825-a35e-4ec6-9aab-069ddfb6108d`

Reply:
Latest commit uses two different prefix to identify the name based and UID based key, and process them accordingly.

```
	// will be removed after v1.5.0
	AnnotationMigratingNamePrefix = prefix + "/migrating-"

	// replace AnnotationMigratingNamePrefix from v1.5.0
	AnnotationMigratingUIDPrefix = prefix + "/migratingUID-"
```